### PR TITLE
Copy connect v7 global database upgrades

### PR DESCRIPTION
**Pull request by avazirna**
## Summary
This PR is to copy Global DB changes that were done as part of the CCC work. The goal with bringing these changes is to ensure a healthy update from that version to production. Changes included here are:
- Added is_local column to ConnectKeyRecord table in global DB

## Safety Assurance

- [X] I have confidence that this PR will not introduce a regression for the reasons below

### Safety story
These changes have already been released in the beta stream, the risk is in the update itself and there are plans to test that update before releasing this to production users.

### CODE REVIEW

File: app/src/org/commcare/models/database/global/GlobalDatabaseUpgrader.java [hunk 1]
```diff
@@ -124,7 +132,44 @@ private boolean upgradeFourFive(SQLiteDatabase db) {
...
+                ConnectKeyRecord newRecord = ConnectKeyRecord.fromV6(oldRecord);
+                //set this new record to have same ID as the old one
+                newRecord.setID(oldRecord.getID());
+                newStorage.write(newRecord);
+            }
+
+            db.setTransactionSuccessful();
+            return true;
+        } catch(Exception e) {
+            return false;
```
Comment (shubham1g5) [inline app/src/org/commcare/models/database/global/GlobalDatabaseUpgrader.java:169]: think we should not be catching exceptions here.

**Review by shubham1g5 — CHANGES_REQUESTED** [review]

File: app/src/org/commcare/models/database/global/GlobalDatabaseUpgrader.java (hunk 1 above)
Comment (avazirna) [reply app/src/org/commcare/models/database/global/GlobalDatabaseUpgrader.java:169]: True. @OrangeAndGreen any particular reason for not letting this bubble up? We might not be able to pick this up if not.

Comment (shubham1g5) [reply app/src/org/commcare/models/database/global/GlobalDatabaseUpgrader.java:169]: Think it was a mistake and Dave is planning to remove it from Connect branch as well

Comment (OrangeAndGreen) [reply app/src/org/commcare/models/database/global/GlobalDatabaseUpgrader.java:169]: Yeah, Shubham is correct, we shouldn't catch the exception and should let it bubble up instead.

**Review by shubham1g5 — APPROVED** [review]

**Pull request merged**